### ISSUE TITLE
comma-dangle always-multiline incorrectly reports error for object rest spread

### ISSUE THREAD

**Issue by damonmaria** [eslint/eslint#7463]
**Tell us about your environment**
ESLint: 3.8.1
Node: 6.7.0
npm: 3.10.8
parser: babel-eslint

**Please show your full configuration:**
```
parser: babel-eslint
extends:
- airbnb
env:
  mocha: true
  meteor: true
rules:
  max-len:
  - warn
  - code: 120
    ignoreTrailingComments: true
    ignoreUrls: true
  padded-blocks: 0
  semi:
  - error
  - never
  space-unary-ops:
  - error
  - overrides:
      "!": true
  quote-props: 0
  no-confusing-arrow: 0
  no-unused-expressions: 0
  no-underscore-dangle: 0
  new-cap:
  - error
  - capIsNewExceptionPattern: "^Match\\.."
  no-param-reassign:
  - error
  - props: false
  no-trailing-spaces: 0
  import/named: error
  import/no-extraneous-dependencies: 0
  import/prefer-default-export: 0
  import/no-unresolved:
  - error
  - ignore:
    - "^redux-form-material-ui$"
  react/prop-types: 0
  react/no-did-mount-set-state: 0
ecmaFeatures:
  spread: true
settings:
  import/resolver:
    meteor:
      extensions:
      - ".js"
      - ".jsx"
```

**What did you do? Please include the actual source code causing the issue.**
The following code causes a `comma-dangle` error on the `...c,` in the function parameter. But not in the object expression that is the result of the function. 

```js
/* eslint comma-dangle: ["error", "always-multiline"] */
const a = ({
  b,
  ...c,
}) => ({
  b,
  ...c,
})
```

**What did you expect to happen?**
With `always-multiline` comma-dangle should not cause an error for anything in this code block.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  17:7   error  Unexpected trailing comma               comma-dangle
```

**Comment by not-an-aardvark** [comment]
Hi,

Thanks for yhe report. However, this is working as intended; trailing commas after spread properties are a syntax error. To fix it, you should remove the trailing comma so that your syntax is valid.

Also see: https://github.com/eslint/eslint/issues/7422, https://github.com/eslint/eslint/issues/7297

**Comment by damonmaria** [comment]
OK, rest spread is always the last param so makes sense. I did search for existing issues but didn't find anything. Apologies.
